dbt/adapters/databricks/api_client.py

```diff
@@ -238,170 +84,88 @@ class CommandExecution(object):
-        if submit_response.status_code != 200:
-            raise DbtRuntimeError(f"Error creating python run.\n {submit_response.content!r}")
+class JobRunsApi:
+    def __init__(self, client: WorkspaceClient, timeout: int):
+        self.client = client
+        self.timeout = timeout
 
-        logger.info(f"Job submission response={submit_response.content!r}")
-        return submit_response.json()["run_id"]
+    def submit(self, run_name: str, job_spec: Dict[str, Any]) -> int:
+        waiter = self.client.jobs.submit(run_name=run_name, tasks=[SubmitTask.from_dict(job_spec)])
+        return waiter.run_id
 
-    def poll_for_completion(self, run_id: str) -> None:
-        self._poll_api(
-            url="/get",
-            params={"run_id": run_id},
-            get_state_func=lambda response: response.json()["state"]["life_cycle_state"],
-            terminal_states={"TERMINATED", "SKIPPED", "INTERNAL_ERROR"},
-            expected_end_state="TERMINATED",
-            unexpected_end_state_func=self._get_exception,
+    def poll_for_completion(self, run_id: int) -> None:
+        logger.debug(f"Waiting on run {run_id}")
+        self.client.jobs.wait_get_run_job_terminated_or_skipped(
+            run_id, timedelta(seconds=self.timeout)
         )
 
-    def _get_exception(self, response: Response) -> None:
-        response_json = response.json()
-        result_state = response_json["state"]["life_cycle_state"]
-        if result_state != "SUCCESS":
-            try:
-                task_id = response_json["tasks"][0]["run_id"]
-                # get end state to return to user
-                run_output = self.session.get("/get-output", params={"run_id": task_id})
-                json_run_output = run_output.json()
-                raise DbtRuntimeError(
-                    "Python model failed with traceback as:\n"
-                    "(Note that the line number here does not "
-                    "match the line number in your code due to dbt templating)\n"
-                    f"{json_run_output['error']}\n"
-                    f"{utils.remove_ansi(json_run_output.get('error_trace', ''))}"
-                )
-
-            except Exception as e:
-                if isinstance(e, DbtRuntimeError):
-                    raise e
-                else:
-                    state_message = response.json()["state"]["state_message"]
-                    raise DbtRuntimeError(
-                        f"Python model run ended in state {result_state} "
-                        f"with state_message\n{state_message}"
-                    )
-
-    def cancel(self, run_id: str) -> None:
+    def cancel(self, run_id: int) -> None:
         logger.debug(f"Cancelling run id {run_id}")
-        response = self.session.post("/cancel", json={"run_id": run_id})
-
-        if response.status_code != 200:
-            raise DbtRuntimeError(f"Cancel run {run_id} failed.\n {response.content!r}")
+        self.client.jobs.cancel_run_and_wait(run_id)
```

Comment:
Do we need to stick to the old way which is cancel Async? https://docs.databricks.com/api/workspace/jobs/cancelrun